lms/services/vitalsource/service.py

```diff
@@ -102,8 +102,13 @@ def get_book_reader_url(cls, document_url) -> str:
         :param document_url: `vitalsource://` type URL identifying the document
         """
         loc = VSBookLocation.from_document_url(document_url)
+        prefix = f"https://hypothesis.vitalsource.com/books/{loc.book_id}"
 
-        return f"https://hypothesis.vitalsource.com/books/{loc.book_id}/cfi/{loc.cfi}"
+        if loc.cfi:
+            return f"{prefix}/cfi/{loc.cfi}"
+        if loc.page:
+            return f"{prefix}/page/{loc.page}"
+        return prefix  # pragma: nocover
```

Comment:
This is unreachable since `from_document_url` throws if the URL has neither a CFI nor a page number.